lib/util.js

```diff
@@ -478,7 +478,19 @@ function formatValue(ctx, value, recurseTimes) {
   if (ctx.showHidden) {
     keys = Object.getOwnPropertyNames(value);
   } else {
-    keys = Object.keys(value);
+    // This might throw if `value` is a Module Namespace Object from an
+    // unevaluated module, but we don't want to perform the actual type
+    // check because it's expensive.
+    try {
+      keys = Object.keys(value);
+    } catch (err) {
+      if (err instanceof ReferenceError) {
```

Comment:
Instead of relying on the reference error here, we might just test for types.isModuleNamespaceObject nevertheless.